### PR TITLE
Add parameter to skip vendor directories

### PR DESCRIPTION
**Pull request by smacker**
In golang it's common practice to commit vendor directory. In other languages, people do it sometimes too.

It messes up results of analysis a lot. This option allows to skip files which start with most popular prefixes `vendor/`, `vendors/`, `node_modules/`.

### CODE REVIEW

File: tree_diff.go [hunk 1]
```diff
@@ -13,13 +14,19 @@ import (
...
 
 const (
 	// DependencyTreeChanges is the name of the dependency provided by TreeDiff.
 	DependencyTreeChanges = "changes"
+	// ConfigTreeDiffSkipVendor is the name of the configuration option
+	// (TreeDiff.Configure()) which allows to skip vendor directories.
+	ConfigTreeDiffSkipVendor = "TreeDiff.SkipVendor"
 )
 
+var vendorDirs = []string{"vendor/", "vendors/", "node_modules/"}
```
Comment (vmarkovtsev) [inline tree_diff.go]: I suggest to move this list to the command line flag (comma separated). For sure there are other directories in other languages. We could name it "BlacklistedDirectories"

Comment (smacker) [reply tree_diff.go]: sorry, could you clarify a bit?

Do you propose to add `BlacklistedDirectories` flag instead of `skip-vendor`? Or add additional flag `BlacklistedDirectories`?

The problem with replacement - it should be empty by default. So the user must know all possible vendor directories if he/she wants to skip them. Not very user-friendly 😞

Comment (vmarkovtsev) [reply tree_diff.go]: I see. Then two flags - rename vendor to blacklist and add `--backlisted-dirs`

**Comment by vmarkovtsev** [comment]
@smacker Let's finish this PR!

finish

**Comment by smacker** [comment]
oh. sorry. I absolutely forgot about it! I'll make changes tonight.

**Comment by smacker** [comment]
Updated with 2 new flags:
  * `--skip-blacklist`
 * `--blacklisted-dirs`

File: tree_diff.go [hunk 2]
```diff
@@ -13,13 +14,22 @@ import (
 // TreeDiff is a PipelineItem.
 type TreeDiff struct {
 	previousTree *object.Tree
+	SkipDirs     []string
```
Comment (vmarkovtsev) [inline tree_diff.go]: Please put it before the private fields

File: tree_diff.go [hunk 3]
```diff
@@ -13,13 +14,22 @@ import (
...
 }
 
 const (
 	// DependencyTreeChanges is the name of the dependency provided by TreeDiff.
 	DependencyTreeChanges = "changes"
+	// ConfigTreeDiffSkipBlacklist is the name of the configuration option
+	// (TreeDiff.Configure()) which allows to skip blacklist directories.
+	ConfigTreeDiffSkipBlacklist = "TreeDiff.SkipVendor"
```
Comment (vmarkovtsev) [inline tree_diff.go]: I would replace "skip" with "enable"

File: tree_diff.go [hunk 4]
```diff
@@ -13,13 +14,22 @@ import (
...
 type TreeDiff struct {
 	previousTree *object.Tree
+	SkipDirs     []string
 }
 
 const (
 	// DependencyTreeChanges is the name of the dependency provided by TreeDiff.
 	DependencyTreeChanges = "changes"
+	// ConfigTreeDiffSkipBlacklist is the name of the configuration option
+	// (TreeDiff.Configure()) which allows to skip blacklist directories.
```
Comment (vmarkovtsev) [inline tree_diff.go]: blacklist -> blacklisted

File: tree_diff.go [hunk 5]
```diff
@@ -13,13 +14,22 @@ import (
...
+	ConfigTreeDiffSkipBlacklist = "TreeDiff.SkipVendor"
+	// ConfigTreeDiffBlacklistedDirs s the name of the configuration option
+	// (TreeDiff.Configure()) which allows to set blacklist directories.
```
Comment (vmarkovtsev) [inline tree_diff.go]: blacklist -> blacklisted

File: tree_diff.go [hunk 6]
```diff
@@ -42,11 +52,27 @@ func (treediff *TreeDiff) Requires() []string {
 
 // ListConfigurationOptions returns the list of changeable public properties of this PipelineItem.
 func (treediff *TreeDiff) ListConfigurationOptions() []ConfigurationOption {
-	return []ConfigurationOption{}
+	options := [...]ConfigurationOption{{
+		Name:        ConfigTreeDiffSkipBlacklist,
+		Description: "Skip blacklist directories.",
+		Flag:        "skip-blacklist",
```
Comment (vmarkovtsev) [inline tree_diff.go]: Same here, IMHO --enable-blacklist is sexier

File: tree_diff.go [hunk 7]
```diff
@@ -42,11 +52,27 @@ func (treediff *TreeDiff) Requires() []string {
 
 // ListConfigurationOptions returns the list of changeable public properties of this PipelineItem.
 func (treediff *TreeDiff) ListConfigurationOptions() []ConfigurationOption {
-	return []ConfigurationOption{}
+	options := [...]ConfigurationOption{{
+		Name:        ConfigTreeDiffSkipBlacklist,
+		Description: "Skip blacklist directories.",
```
Comment (vmarkovtsev) [inline tree_diff.go]: blacklist -> blacklisted

File: tree_diff.go [hunk 8]
```diff
@@ -42,11 +52,27 @@ func (treediff *TreeDiff) Requires() []string {
...
+		Type:        StringsConfigurationOption,
+		Default:     defaultBlacklistedDirs},
+	}
+	return options[:]
 }
 
 // Configure sets the properties previously published by ListConfigurationOptions().
-func (treediff *TreeDiff) Configure(facts map[string]interface{}) {}
+func (treediff *TreeDiff) Configure(facts map[string]interface{}) {
+	if val, exists := facts[ConfigTreeDiffSkipBlacklist]; exists && val.(bool) == true {
```
Comment (vmarkovtsev) [inline tree_diff.go]: We can safely shorten this to `&& val.(bool)`

**Comment by vmarkovtsev** [comment]
Appveyor build failed because of the recent Chocolatey  update, `choco install` now requires `--allow-empty-checksums`. Could you please add this flag to appveyor.yaml

**Comment by smacker** [comment]
Boring renaming is done. (also I like `skip-blacklist` more than `enable-blacklist` because it actually tells what would happen when you *enable* but whatever)
Flag for appveyor also.

**Comment by vmarkovtsev** [comment]
Tests failed

File: tree_diff.go [hunk 9]
```diff
@@ -42,11 +52,27 @@ func (treediff *TreeDiff) Requires() []string {
 
 // ListConfigurationOptions returns the list of changeable public properties of this PipelineItem.
 func (treediff *TreeDiff) ListConfigurationOptions() []ConfigurationOption {
-	return []ConfigurationOption{}
+	options := [...]ConfigurationOption{{
+		Name:        ConfigTreeDiffEnableBlacklist,
+		Description: "Enable blacklisted directories.",
```
Comment (vmarkovtsev) [inline tree_diff.go]: This is a bit misleading... "Skip blacklisted directories" is better. And yep, now I see the problem, lol, please revert it back to your name `--skip-backlist`

**Comment by smacker** [comment]
The option is renamed, but internal variables are kept the same.

P.S. It's weird that codecov shows less coverage. Last commits don't add any code, only rename. And before codecov was happy...
